Documentation/proposals/202405-docs-restructuring.md

```diff
@@ -0,0 +1,104 @@
+
+Currently, there is some unnecessary information from the documentation creating misconceptions in the user's mind. For example, let us look at [#6046](https://github.com/prometheus-operator/prometheus-operator/issues/6046) which tells us that there is no Ingress Guide present in the current documentation. But, if we look at the website, there are links to the **“Ingress Guide”** on the [Getting-Started](https://prometheus-operator.dev/docs/user-guides/getting-started/#exposing-the-prometheus-service) and [Alerting page](https://prometheus-operator.dev/docs/user-guides/alerting/#exposing-the-alertmanager-service) in **User-Guide**. Due to this, many users will report the same issue and it will take time for maintainers to resolve them.
+
+Incorporation of new topics is also difficult if the structure is not up to mark because more time and effort is needed to decide the best place to add a topic which can often lead to decrease in productivity of a maintainer. For example, in issue [#3553](https://github.com/prometheus-operator/prometheus-operator/issues/3553#issuecomment-726733177), it has been mentioned that basic architecture needs to be worked upon before adding the diagram as there is no section talking about **“namespace selection”** in the current documentation.
+
+# Goals
+
+- To improve readability and make the documentation easier to understand.
+- To ensure a uniform flow of content.
+- To remove irrelevant information from the documentation.
+- To ensure that all new features get documented properly in the future.
+
+# Audience
+
+- Developers who rely on Prometheus Operator to facilitate the deployment, management and monitoring of Prometheus instances in their environment.
+- Developers who want to set up alerts for their application.
+- Administrators who want to monitor Prometheus and Alertmanager instances using Kubernetes tools.
+- Maintainers who are responsible for optimizing Prometheus Operator.
+- Contributors who want to engage and want to contribute effectively to the project.
+
+# Non-Goals
+
+- Adding extra documentation about new features, enhancements, troubleshooting, or other information that might be missing today.
+
+# How
+
+I believe a basic structure for the website should be as given below:
+
+### 1. Getting Started
+
+- **Introduction** - This section will introduce us to Prometheus-Operator and will talk about goals and the problems that are being solved by this project.
+- **Installation Guide** - This section will contain all the methods of installation of Prometheus-Operator in the production environment.
+- **Compatibility** - This section will provide information about compatibility of Prometheus-Operator with Kubernetes, Prometheus, Alertmanager and Thanos.
+- **Design** - This section will describe the design of Prometheus-Operator focusing on the various custom resource definitions(CRDs) it manages.
```

Comment:
"Design" is a bit overloaded and I'm not sure that it's relevant for a Getting Started section?

Reply:
@AshwinSriram11 please confirm this, but I think the intention was to move [this page](https://prometheus-operator.dev/docs/operator/design/) somewhere, and this page looks relevant to all personas.

Reply:
We discussed to move the Design page to the "Getting Started" section.The reason behind this was that after installation, a user can get information about all the CRDs offered by Prometheus-Operator before moving on to "how to use them". Since it is relevant to all persona and is a one page overview, this is why I thought it should be in the "Getting Started" section.

Reply:
Comparing the terse [description](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.Prometheus) of the Prometheus CRD with the [longer one](https://prometheus-operator.dev/docs/operator/design/#prometheus) from the Design page, I don't see why we would not copy the latter to the Go code?